### PR TITLE
Refactor Version class

### PR DESCRIPTION
**Pull request by RoiEXLab**
Basically https://github.com/triplea-game/triplea/pull/2470 but "without functional changes" (the only difference is that version strings are now printed in the normal 1.9.0.0.build scheme)

### CODE REVIEW

File: src/main/java/games/strategy/util/Version.java [hunk 1]
```diff
@@ -158,29 +165,44 @@ private int compareTo(final Version other, final boolean ignoreMicro) {
     return 0;
   }
 
+  /**
+   * Returns true, if the provided Version object is greater than this object.
```
Comment (ssoloff) [inline src/main/java/games/strategy/util/Version.java]: I think the Javadocs for `isGreaterThan()` and `isLessThan()` are reversed.  `isGreaterThan()` appears to return `true` if **this** `Version` is greater than the specified `Version`.  Similarly, `isLessThan()` appears to return `true` if **this** `Version` is less than the specified `Version`.

For example, the following tests pass:

```java

  @Test
  public void isGreaterThan() {
    assertTrue(new Version(2, 0).isGreaterThan(new Version(1, 0)));
    assertFalse(new Version(1, 0).isGreaterThan(new Version(2, 0)));
    assertFalse(new Version(1, 0).isGreaterThan(new Version(1, 0)));
  }

  @Test
  public void isLessThan() {
    assertFalse(new Version(2, 0).isLessThan(new Version(1, 0)));
    assertTrue(new Version(1, 0).isLessThan(new Version(2, 0)));
    assertFalse(new Version(1, 0).isLessThan(new Version(1, 0)));
  }
```

Comment (simon33-2) [reply src/main/java/games/strategy/util/Version.java]: So in the first case 2.0 is greater than 1.0 and that test passes. That is what I would expect. Perhaps the javadoc needs to be reworded if it implies something different.

**Comment by simon33-2** [comment]
I don't understand why this is changing?

Why is 1.9.0.0.build "normal"? I don't believe that was ever adopted intentionally. Just tolerated.

**Comment by simon33-2** [comment]
Should we get rid of the out of date engine version warning?

File: src/main/java/games/strategy/util/Version.java (hunk 1 above)
Comment (ssoloff) [reply src/main/java/games/strategy/util/Version.java]: Yup, all I was suggesting is to update the Javadocs for `isGreaterThan(Version)`, `isGreaterThan(Version, boolean)`, and `isLessThan(Version)`.  Adding test cases for these methods would be a bonus. :smile:

Comment (RoiEXLab) [reply src/main/java/games/strategy/util/Version.java]: @ssoloff Thanks for the tests (and suggestion of course, I was assuming there were already tests for such a trivial thing)

Comment (ssoloff) [reply src/main/java/games/strategy/util/Version.java]: @RoiEXLab Sorry to nit on this, but with the most recent changes to the Javadocs, you also need an "or equal to" clause.  For example, for `isGreaterThan()`:

<pre>
/**
 * Returns true, if the provided Version object is less than <b>or equal to</b> this object.
 */
</pre>

I actually thought you were just going to swap the parameters around in the description:

```
/**
 * Returns true, if this object is greater than the provided Version object.
 */
```

But either way works. :smile:

Comment (RoiEXLab) [reply src/main/java/games/strategy/util/Version.java]: @ssoloff And that is why you should code when you are tired in general ^^
Of course the latter case is easier to understand, I re-did the commit...

**Pull request merged**